### ISSUE TITLE
Tests: when run in group, make them NOT share database

### ISSUE THREAD

**Issue by alebg** [dream-aim-deliver/kernel-planckster#49]
When testing in group, the tests will see dangling entries in their database coming from other tests. This makes testing of cases when a table should be empty impossible. See for example the following test for ConversationRepository list_conversations, made for the case when there are *no* conversations. This cannot be put right now in the test files because if run in group (and in the CI) it fails. I'll preserve this test code here, but for now it won't be put in the code:

```python
def test_error_list_conversations_no_conversations(app_container: Container, db_session: TDatabaseFactory) -> None:
    conversation_repository = app_container.sqla_conversation_repository()

    list_convs_DTO: ListConversationsDTO = conversation_repository.list_conversations()

    assert list_convs_DTO.status == False
    assert list_convs_DTO.errorCode == -1
    assert list_convs_DTO.errorMessage == "No Conversations found in the database."
    assert list_convs_DTO.errorName == "No Conversations found"
    assert list_convs_DTO.errorType == "NoConversationsFound"
```

**Comment by maany** [comment]
The correct way to fix this is to clean up after the tests. The database changes made by a test should be reverted by the test itself at the end of the test run. This way a test always leaves a fresh database to be re-used by the next test in the chain. 
If tests are run in parallel, we can still run into the issue you are describing, however, there are ways to manage that too ( currently out of scope).
If everything in the tests is already using a fixture, we can change these fixtures to yield fixtures and they can automatically clean up after the test runs. This would require a bit of refactoring as you need a contextmanager to work with yield fixtures.
